lib/rubygems/version.rb

```diff
@@ -170,7 +170,8 @@ def version
   # True if the +version+ string matches RubyGems' requirements.
 
   def self.correct? version
-    return false if version.nil?
+    warn "nil versions are discouraged and will be deprecated in Rubygems 4" if version.nil?
```

Comment:
should this use `Gem::Deprecate` ?

Reply:
Should the year and month be? "2018", "12"

Reply:
ah yeah, always forget that's how it works :/

In that case, only warn `unless Gem::Deprecate.skip?` ?

Reply:
done